e3sm_diags/e3sm_diags_driver.py

```diff
@@ -408,6 +409,8 @@ def main(parameters=[]):
         )
         raise Exception(message)
 
+    return parameters_results
```

Comment:
`parameter_results` are now returned so that we can parse them for the `results_dir` in unit/integration tests.

It is also generally useful to see the results after each diagnostic run.